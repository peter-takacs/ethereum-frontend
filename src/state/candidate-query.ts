
export interface CandidateQuery {
    candidate?: string,
    certificate?: string
}