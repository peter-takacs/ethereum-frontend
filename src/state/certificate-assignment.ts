export enum CertificateQueryState {
    Idle,
    Loading,
    Assigned,
    NotAssigned
}

export interface State {
    candidate: string,
    certificate: string,
    state: CertificateQueryState
}