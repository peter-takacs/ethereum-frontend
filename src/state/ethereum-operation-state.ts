export enum EthereumOperationState {
    Idle,
    InProgress,
    Completed,
    Error
}