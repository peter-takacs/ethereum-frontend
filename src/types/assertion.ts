import { Address } from "./ethereum-address";

export interface Assertion {
    certificate: string;
    issuer: Address;
}