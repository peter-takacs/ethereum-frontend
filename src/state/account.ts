import { Address } from "../types/ethereum-address";

export interface AccountState {
    address: Address | null;
}