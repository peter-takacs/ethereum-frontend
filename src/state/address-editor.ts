import { Address } from "../types/ethereum-address";

export interface State {
    placeholder?: string;
    address: Address | null
}