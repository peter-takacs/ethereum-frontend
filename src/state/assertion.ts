import { Address } from "../types/ethereum-address";

export interface Assertion {
    certificate: string;
    issuer: Address;
}