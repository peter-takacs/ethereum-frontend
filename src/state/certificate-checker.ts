import { Address } from "../types/ethereum-address";

export enum CertificateCheckState {
    Idle,
    Checking,
    Confirmed,
    Rejected
}
export interface State {
    candidate: Address | null,
    certificate: string,
    certificateStatus: CertificateCheckState
}