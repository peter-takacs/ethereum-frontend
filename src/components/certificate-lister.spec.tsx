import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CertificateChecker from './certificate-checker';
import { dummyAddress } from '../utils/test/address';
import CertificateLister from './certificate-lister';
import { EthereumOperationState } from '../state/ethereum-operation-state';
import { Assertion } from '../state/assertion';

test('Renders correctly in every state without results', () => {
    [
        EthereumOperationState.Idle,
        EthereumOperationState.Completed,
        EthereumOperationState.Completed,
        EthereumOperationState.Error
    ].forEach(state => {
        const component = renderer.create(
            <CertificateLister
                candidate={dummyAddress(0)}
                onCandidateChange={() => { }}
                onSubmit={() => { }}
                state={state}
                results={[] as Assertion[]} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})

test('Renders correctly with results', () => {
    const dummyAssertions = [
        {certificate: 'test', issuer: dummyAddress(0)},
        {certificate: 'foo', issuer: dummyAddress(1)},
    ]
    const component = renderer.create(
        <CertificateLister
            candidate={dummyAddress(0)}
            onCandidateChange={() => { }}
            onSubmit={() => { }}
            state={EthereumOperationState.Completed}
            results={dummyAssertions} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
