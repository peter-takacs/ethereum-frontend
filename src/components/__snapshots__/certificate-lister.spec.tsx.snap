// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly in every state without results 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value="0xdf3f619804a92fdb4057192dc43dd748ea778adc"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-144"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Result
  </div>
</div>
`;

exports[`Renders correctly in every state without results 2`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value="0xdf3f619804a92fdb4057192dc43dd748ea778adc"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-144"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Result
  </div>
</div>
`;

exports[`Renders correctly in every state without results 3`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value="0xdf3f619804a92fdb4057192dc43dd748ea778adc"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-144"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Result
  </div>
</div>
`;

exports[`Renders correctly in every state without results 4`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value="0xdf3f619804a92fdb4057192dc43dd748ea778adc"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-144"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Result
  </div>
</div>
`;

exports[`Renders correctly with results 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value="0xdf3f619804a92fdb4057192dc43dd748ea778adc"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-144"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Result
  </div>
</div>
`;
